omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java

```diff
@@ -65,6 +77,39 @@ public Attachment getByUniqueId(String uniqueId) {
 		}
 	}
 	
+	/**
+	 * Gets attachements by patient or encounter (paged according to context if necessary) only if a
+	 * patient or encounter parameter exists respectively in the request set on the
+	 * {@link RequestContext} otherwise searches for obs that match the specified query
+	 *
+	 * @param context
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String patientUuid = context.getRequest().getParameter("patient");
+		if (patientUuid != null) {
+			Patient patient = ((PatientResource1_8) Context.getService(RestService.class)
+			        .getResourceBySupportedClass(Patient.class)).getByUniqueId(patientUuid);
+			if (patient == null)
```

Comment:
So this will go. Btw **always** use enclosing brackets with an `if`, even for a one liner.